src/hotspot/share/gc/shenandoah/shenandoahLock.cpp

```diff
@@ -28,9 +28,47 @@
 
 #include "gc/shenandoah/shenandoahLock.hpp"
 #include "runtime/atomic.hpp"
+#include "runtime/interfaceSupport.inline.hpp"
 #include "runtime/javaThread.hpp"
 #include "runtime/os.inline.hpp"
 
+// These are inline variants of Thread::SpinAcquire with optional blocking in VM.
+
+class ShenandoahNoBlockOp : public StackObj {
+public:
+  ShenandoahNoBlockOp(JavaThread* java_thread) {
```

Comment:
Should this be inline? Or would compiler be clever enough to not emit anything there?
Edit: or doesn't it matter, because we are going to yield anyway?

Reply:
I don't think it matters, because the actual yield would dominate. Also, I don't think we do `inline` for RAII constructors...